### DISCUSSION TITLE
In getServerSideProps, is it possible to know if the request is from the client or server?

### DISCUSSION THREAD

**Discussion by rorhug** [vercel/next.js#22315] [Help]
We use `getServerSideProps` to warm the cache for apollo based on the hooks on the page. We want to skip that procedure if the page is being rendered on client. 

Is it possible to know via context whether the request is happening on the client?

**Comment by julien-amblard** [comment]
Hello, I also was looking for that because I want to display the page with placeholder while data is fetching, I wasn't a big fan about freezing the current page while data is loading server side and then displaying the page. IMO it's a weird behaviour of Next.

There is the [preview mode](https://nextjs.org/docs/advanced-features/preview-mode) in the documentation, but I'm still not a big fan of that, but maybe I just don't understand the right use of it.

Anyway, the only solution I've find is to check the `context.req.url`, when we are server side (first ping) the called url is something like that `/my/url/` but when we are client side (navigating through the app) it's a json call `/my/url.json`, so with a simple regexp `url.match(/.json/)`  you can know the context

It's not a perfect solution, but it work, I would be glad to have a boolean prop in the context to directly know that 🌞

Hope it help, if someone find a cleaner solution I'll take it !

**Reply by rorhug** [reply]
Oh Awesome thanks - i'll try that now!
